gopher/annotations.py

```diff
@@ -146,25 +154,48 @@ def load_annotations(species, aspect="all", release="current", fetch=False):
     species = SPECIES.get(species.lower(), species.lower())
     annot_file = download_annotations(species, release=release, fetch=fetch)
 
-    polars_annot = pl.read_csv(
+    keep = [
+        "uniprot_accession",
+        "go_id",
+        "aspect",
+        "db",
+        "gene_product_form_id",
+    ]
+    polars_annot = pl.scan_csv(
         annot_file,
         separator="\t",
         comment_char="!",
         has_header=False,
         new_columns=cols,
     )
-    annot = polars_annot.to_pandas()
+    polars_annot = polars_annot.select(keep)
 
     if aspect is not None:
-        annot = annot.loc[annot["aspect"] == aspect, :]
+        polars_annot = polars_annot.filter(pl.col("aspect") == aspect)
+
+    polars_annot = polars_annot.with_columns(
+        [
+            pl.when(pl.col("db").eq("UniProtKB"))
+            .then(pl.col("uniprot_accession"))
+            .otherwise(
+                pl.col("gene_product_form_id")
+                .fill_null("")
+                .str.extract(r"UniProtKB:(.+)", 1)
+            )
+            .alias("uniprot_accession")
+        ],
+    )
+
+    polars_annot = (
+        polars_annot.select(["uniprot_accession", "go_id", "aspect"])
+        .unique()
+        .with_columns(pl.col("go_id").map_dict(terms).alias("go_name"))
+    )
 
-    uniprot = annot["db"] == "UniProtKB"
-    if not uniprot.all():
-        annot.loc[~uniprot, "uniprot_accession"] = annot.loc[
-            ~uniprot, "gene_product_form_id"
-        ].str.extract(r"UniProtKB:(.+)", expand=False)
+    annot = polars_annot.collect().to_pandas()
 
-    keep = ["uniprot_accession", "go_id", "aspect"]
-    annot = annot.loc[:, keep].drop_duplicates()
-    annot["go_name"] = annot["go_id"].map(terms)
     return annot, mapping
+
+
+if __name__ == "__main__":
+    load_annotations("human", aspect="all", release="current", fetch=False)
```

Comment:
I don't think we need this? We load annotations the first time a function requiring annotations is called.

Reply:
You are right, I just used this during testing locally and forgot totake it out :P